test/cases/BaseDeployMissionUnit.test.ts

```diff
@@ -0,0 +1,214 @@
+          `0xcontractAddress_Contract_${mission}`
+        );
+        // does it make sense to do this?
+        assert.deepEqual(
+          await campaign.state.instances[mission].deployArgs(),
+          [`arg_${mission}1`, `arg_${mission}2`]
+        );
+      }
+    });
+
+    it("#savetoDB() Should call `saveToDB()` when deploy a contract", async () => {
+      for (const mission of missionIdentifiers) {
+        if (mission !== "deployed") {
+          assert.equal(
+            // ts complains about `called` prop
+            // @ts-ignore
+            await campaign.state.instances[mission].called.includes("saveToDB"),
+            true
+          );
+        }
+      }
+    });
+  });
+
+  describe("Minor methods", () => {
+    it("#deployArgs() Should return correct deploy arguments", async () => {
+      for (const mission of missionIdentifiers) {
+        await campaign.state.instances[mission].deployArgs();
+      }
+    });
+
+    it("#updateStateContract() Should update state contract when deploys the contracts", async () => {
+      const state = await campaign.state.contracts;
+
+      for (const mission of missionIdentifiers) {
+        const contract = state[mission];
+
+        assert.strictEqual(
+          typeof contract.deploymentTransaction,
+          "function",
+          "Not a contract. Method 'deploymentTransaction' should exist"
+        );
+
+        assert.strictEqual(
+          typeof contract.getAddress,
+          "function",
+          "Not a contract. Method 'getAddress' should exist"
+        );
+
+        assert.strictEqual(
+          typeof contract.interface,
+          "object",
+          "Not a contract. Property 'interface' should exist"
+        );
+
+        assert.strictEqual(
+          typeof contract.target,
+          "string",
+          "Not a contract. Property 'target' should exist and be an address"
+        );
+
+        assert.strictEqual(
+          typeof contract.waitForDeployment,
+          "function",
+          "Not a contract. Function 'waitForDeployment' should exist"
+        );
+      }
+    });
+
+    it("#buildObject() Should build correct object of contract and call insertOne()", async () => {
+      const {
+        buildObject,
+      } = campaign.state.instances;
+
+      const {
+        buildObject: contractBuildObject,
+      } = campaign.state.contracts;
+
+      const buildedDbObject = await buildObject.buildDbObject(contractBuildObject, buildObject.implAddress);
+
+      const resultBuildedDbObject = {
+        address: "0xcontractAddress_Contract_buildObject",
+        abi: "[]",
+        bytecode: "0xbytecode",
+        implementation: null,
+        name: "Contract_buildObject",
+      };
+
+      assert.deepEqual(
+        buildedDbObject,
+        resultBuildedDbObject
+      );
+    });
+  });
+
+  describe("#needsDeploy()",() => {
+    it("Should return FALSE, because it found itself in db", async () => {
```

Comment:
these tests need better titles.
Also, when `false` is returned from `needsDeploy()` the current contract needs to be written into the state from the DB. not checked here